File: 18-temperature-converter/temperature_converter.py
to_celsius = input("Press C to convert from Fahrenheit to Celsius.\n"
                   "Press F to convert from Celsius to Fahrenheit.\n"
                   "Your choice: ").lower() == "c"
temperature = float(input(f"Please enter the temperature in {'Fahrenheit' if to_celsius else 'Celsius'}: "))
if to_celsius:
    converted_temperature = (temperature - 32) * (5 / 9.0)
else:
    converted_temperature = (temperature * (9 / 5.0)) + 32
print(f"The temperature in {'Celsius' if to_celsius else 'Fahrenheit'}",
      f"is {converted_temperature:.1f}.")

File: 18-temperature-converter/TemperatureConverter.java
class TemperatureConverter {
  public static void main(String[] args) {
    var line = System.console().readLine("Press C to convert from Fahrenheit to Celsius.%nPress F to convert from Celsius to Fahrenheit.%nYour choice: ");
    var toCelsius = line.toLowerCase().equals("c");

    line = System.console().readLine("Please enter the temperature in " + (toCelsius ? "Fahrenheit" : "Celsius") + ": ");
    double temperature = Double.parseDouble(line);

    double convertedTemperature;
    if (toCelsius) {
      convertedTemperature = (temperature - 32) * (5 / 9.0);
    } else {
      convertedTemperature = (temperature * (9 / 5.0)) + 32;
    }

    System.out.printf("The temperature in %s is %.2f.%n", toCelsius ? "Celsius" : "Fahrenheit", convertedTemperature);
  }
}
